### PR TITLE
Add Currency Selection

### PR DESCRIPTION
**Pull request by blurpesec**
# Description
Adds currency selection to settings page and updates `getRate` && `getAssetRate` methods to pull those rates and display on dashboard, and throughout the app :-)

To start, added support for USD, EUR, GBP, TRY (Turkish Lira), CNY (Chinese Yuan), INR (Indian Rupee), and RUB (Russian Rubles). I defaulted to showing the asset symbols in front of the numbers due to it seeming that the symbol being a prefix/suffix depends on the locality of the user, not necessarily homogeneous per currency.

Here's the currency selection:
image

Here's an example of balance breakdown:
image

### CODE REVIEW

**Comment by LuckyLuker** [comment]
[ch177]

File: common/v2/components/TransactionFlow/TxReceipt.tsx [hunk 1]
```diff
@@ -346,7 +352,10 @@ export const TxReceiptUI = ({
             <AssetIcon uuid={asset.uuid} size={'24px'} />
             <Amount
               assetValue={`${parseFloat(assetAmount()).toFixed(6)} ${assetTicker()}`}
-              fiatValue={`$${convertToFiat(parseFloat(assetAmount()), assetRate()).toFixed(2)}
+              fiatValue={`${Fiats[settings.fiatCurrency].symbol}${convertToFiat(
```
Comment (FrederikBolding) [inline common/v2/components/TransactionFlow/TxReceipt.tsx:355]: Can we do a helper function to find the fiat instead this kind of lookup?

(Same for other files)

File: common/v2/config/fiats.ts [hunk 2]
```diff
@@ -13,12 +13,38 @@ export const USD = {
...
-  symbol: '€' as TSymbol
+  symbol: '€' as TSymbol,
+  prefix: true
 };
 export const GBP = {
   code: 'GBP',
   name: 'British Pounds',
-  symbol: '£' as TSymbol
+  symbol: '£' as TSymbol,
+  prefix: true
```
Comment (FrederikBolding) [inline common/v2/config/fiats.ts:23]: Not sure if prefixing by default is the best option? I think I'm used to seeing it suffixed unless it's USD?

File: common/v2/features/SendAssets/components/SendAssetsForm.tsx [hunk 3]
```diff
@@ -512,9 +515,9 @@ const SendAssetsForm = ({ txConfig, onComplete }: IStepComponentProps) => {
                       values.advancedTransaction ? values.gasPriceField : values.gasPriceSlider
                     }
                     fiatAsset={{
-                      fiat: 'USD',
+                      fiat: Fiats[settings.fiatCurrency].code || 'USD',
```
Comment (FrederikBolding) [inline common/v2/features/SendAssets/components/SendAssetsForm.tsx:518]: Can this ever be undefined?

File: common/v2/features/SendAssets/components/SendAssetsForm.tsx [hunk 4]
```diff
@@ -512,9 +515,9 @@ const SendAssetsForm = ({ txConfig, onComplete }: IStepComponentProps) => {
...
                       rate: (getAssetRate(baseAsset || undefined) || 0).toString(),
-                      symbol: '$'
+                      symbol: Fiats[settings.fiatCurrency].symbol || '$'
```
Comment (FrederikBolding) [inline common/v2/features/SendAssets/components/SendAssetsForm.tsx:520]: Can this ever be undefined?

File: common/v2/components/TransactionFlow/TxReceipt.tsx (hunk 1 above)
Comment (blurpesec) [reply common/v2/components/TransactionFlow/TxReceipt.tsx:355]: sure sure. that'd be good to help with `fiatCurrency of undefined` errs too :)

File: common/v2/features/SendAssets/components/SendAssetsForm.tsx (hunk 3 above)
Comment (blurpesec) [reply common/v2/features/SendAssets/components/SendAssetsForm.tsx:518]: I suppose not :-0

File: common/v2/features/SendAssets/components/SendAssetsForm.tsx (hunk 4 above)
Comment (blurpesec) [reply common/v2/features/SendAssets/components/SendAssetsForm.tsx:520]: I suppose not :-0

File: common/v2/config/fiats.ts (hunk 2 above)
Comment (blurpesec) [reply common/v2/config/fiats.ts:23]: It seems to be based on locality of the user, not on the currency symbol itself. But if it's not actually a symbol, but a string of letters instead you're supposed to post-fix. We don't really have a way to implement the localities though.. soooo?

**Pull request merged**